adhocteam/Head-Start-TTADP
# Import tta plan goals script and schema

**Description of change**
This PR parses TTA plan goals in the .csv format and enters the data into the db. Models and migrations are created to support the import.

**How to test**
Running locally:
- `yarn deps`
- `yarn db:migrate`
- `yarn import:goals`
- verify that the data has been imported

To test using the original smartsheet file:
- Navigate to https://app.smartsheetgov.com/sheets/Whm6xJ5567j288GCf68WxxW7fjFqMVg86rFGxPQ1?view=grid
- Export the file to MS Excel or Google Sheets
- Replace smart quotes with non-smart quotes
- Save as comma delimited CSV, e.g. GranteeTTAPlan.csv
- Follow the above steps

To run the new tests:
- `node ./node_modules/jest/bin/jest tools/importPlanGoals.test.js`

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/57

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
